src/app/static/src/app/apiService.ts

```diff
@@ -4,44 +4,95 @@ import {
     Failure,
     InitialiseModelRunResponse,
     ModelRunResultResponse,
-    Success,
     ValidateInputResponse
 } from "./generated";
+import {Commit} from "vuex";
 
-type ResponseData = ValidateInputResponse | ModelRunResultResponse | InitialiseModelRunResponse | InternalResponse | null
+type ResponseData =
+    ValidateInputResponse
+    | ModelRunResultResponse
+    | InitialiseModelRunResponse
+    | InternalResponse
+    | Boolean
 
-export class APIService {
+declare var appUrl: string;
 
-    get<T extends ResponseData>(url: string) {
-        return axios.get(url)
-            .then((response: AxiosResponse<Success>) => {
-                const payload = response && response.data;
-                return payload.data as T
-            })
-            .catch((e: AxiosError<Failure>) => {
-                return this.handleError(e)
-            })
-    }
+export interface API {
+
+    commitFirstErrorAsType: (commit: Commit, type: string) => API
+    ignoreErrors: () => API
+
+    postAndReturn<T extends ResponseData>(url: string, data: any): Promise<void | T>
+
+    get<T extends ResponseData>(url: string): Promise<void | T>
+}
 
-    postAndReturn<T extends ResponseData>(url: string, data: any) {
-        return axios.post(url, data)
-            .then((response: AxiosResponse<Success>) => {
-                const payload = response && response.data;
-                return payload.data as T
-            })
-            .catch((e: AxiosError<Failure>) => {
-               return this.handleError(e)
-            });
+export class APIService implements API {
+
+    // appUrl will be set as a jest global during testing
+    private readonly _baseUrl = typeof appUrl !== "undefined" ? appUrl : "";
+
+    private _buildFullUrl = (url: string) => {
+        return this._baseUrl + url
+    };
+
+    private _ignoreErrors: Boolean = false;
+
+    static getFirstErrorFromFailure = (failure: Failure) => {
+        const firstError = failure.errors[0];
+        return firstError.detail ? firstError.detail : firstError.error;
+    };
+
+    private _notifyOnError: ((failure: Failure) => void) = (failure: Failure) => {
```

Comment:
We don't ever really want to do this, do we? Could we force the consumers of APIService to either choose to ignore errors or to commit them as mutations? Maybe it's an error if they haven't specified one or the other when they make the get or post? This contradicts my other comment about the method call ordering though!

Reply:
I've gone with a console.warn if no error handler is attached - https://github.com/mrc-ide/hint/pull/28/files#diff-0f0781cfcf5e78654409fce90c2c466fR100